### ISSUE TITLE
Embedded Docs

### ISSUE THREAD

**Issue by sofroniewn** [napari/napari#2589]
## 🚀 Feature
As mentioned by @tlambert03 here https://github.com/napari/napari/issues/2584#issuecomment-823370101 it would be great for napari to have an up to date copy of its docs embedded in the application.

## Motivation

Easy access to in application docs could really help new comers learn about napari's API and how to use napari.

## Additional context

At one point in time @tlambert03 I believe you showed me a prototype for how this might work. Now that we're using jupyterbook I'm curious if you have thoughts on the approach we should take. I'm also curious if @goanpeca @Carreau @ppwadhwa have ideas on how to best do this.

**Comment by goanpeca** [comment]
@sofroniewn is this akin to having a docs viewer panel?

**Comment by tlambert03** [comment]
> is this akin to having a docs viewer panel?

yeah, the idea would just be to grab the static output (`_build/html`) from our docs build and include it in the distribution.  Then provide a builtin browser to use it.  (presumably it would be most useful if we could figure out how to actually link to the relevant objects and pages from within the application).  Do you have any experience with that kind of thing from Spyder @goanpeca?

> Now that we're using jupyterbook I'm curious if you have thoughts on the approach we should take

@sofroniewn I don't think it changes anything... still just browsing html at the end of the day

**Comment by sofroniewn** [comment]
> @sofroniewn is this akin to having a docs viewer panel?

a docs dock widget 🤣

**Comment by goanpeca** [comment]
>Do you have any experience with that kind of thing from Spyder

Yeah, kinda :)

>(presumably it would be most useful if we could figure out how to actually link to the relevant objects and pages from within the application).

Like having tooltips that include a Urls like `Click for more information`? 

>a docs dock widget 🤣

I can work on this :)

**Comment by sofroniewn** [comment]
> Like having tooltips that include a Urls like Click for more information?

Wow, love that idea! Or maybe you "shift-right click" on a button and it opens the help? I don't know I'm just imagining things now.

@liaprins-czi and @LCObus might have good ideas here too!!

**Comment by tlambert03** [comment]
> Yeah, kinda :)

I wonder if perhaps that’s an understatement? 😂 

> Like having tooltips that include a Urls like Click for more information?

exactly!  Or (just dreaming now) having the console `?` Open docs, etc... But yeah, a general “what is this?” Mechanism that links 

> I can work on this :)

yay :)

**Comment by goanpeca** [comment]
>Wow, love that idea! Or maybe you "shift-right click" on a button and it opens the help? I don't know I'm just imagining things now.

Sure, let's explore ideas here :) 

>exactly! Or (just dreaming now) having the console ? Open docs, etc... 

>But yeah, a general “what is this?” Mechanism that links

Sounds like a plan :)

**Comment by goanpeca** [comment]
See https://github.com/napari/napari/pull/2607 for an option of where to put this.

**Comment by sofroniewn** [comment]
> See #2607 for an option of where to put this.

Nice, I've left some more general comments on the side bar over on that PR. I had originally just been thinking of the help in a dock widget - was there a particular reason for thinking of the side bar in this context? I think we'll probably want to have the help in its own window so that one can have it be very wide without forcing the image the be very small. We still might want to use side bars too, but that is just one thought. Curious how others feel too.

**Comment by Carreau** [comment]
One of the question is if you ship it with napari; it will make the installer much bigger. Maybe a plugin that is compatible with a single version of napari and that ships with the the html doc ?

It also depends if you want to also sho non-napari doc in it; in which case you could use https://pypi.org/project/docrepr/.

and in a few month, once it's ready, maybe https://github.com/Carreau/papyri

**Comment by sofroniewn** [comment]
We could have `napari-docs` be a dock widget plugin to start with. I'd be open to that.

> It also depends if you want to also sho non-napari doc in it; in which case you could use https://pypi.org/project/docrepr/.
>
> and in a few month, once it's ready, maybe https://github.com/Carreau/papyri

That would be cool!

**Comment by goanpeca** [comment]
>Nice, I've left some more general comments on the side bar over on that PR. I had originally just been thinking of the help in a dock widget - was there a particular reason for thinking of the side bar in this context? I think we'll probably want to have the help in its own window so that one can have it be very wide without forcing the image the be very small. We still might want to use side bars too, but that is just one thought. Curious how others feel too.

>so that one can have it be very wide without forcing the image the be very small

This is assuming that there will be huge screenshots of full screen napari and this could be reconsidered in the help and what type of images are included.

Sidebars help with things that have quick access but are meant to be transient. I want to install plugin I open the sidebar plugin, look for it, install it, then collapse and move with my work, when the plugin finishes (failing or succeeding) the notification manager will let me know.

I see the same happening with help, either by right clicking on any part on napari and having the `what is this?` open the sidebar with help, reading, searching for stuff and then collapsing it again, but it is a quick access thing.

Layers could also be a sidebar thing but then again, the question is about is this something I sometime want to hide?

**Comment by goanpeca** [comment]
> and in a few month, once it's ready, maybe https://github.com/Carreau/papyri

Rooting for this 🚀

**Comment by tlambert03** [comment]
> it will make the installer much bigger.

very true.  Our wheel is ~1.4MB ... our built docs are ~15MB.  So this might be something we only want to do for the bundled app?

**Comment by sofroniewn** [comment]
> very true. Our wheel is ~1.4MB ... our built docs are ~15MB. So this might be something we only want to do for the bundled app?

what about a plugin that gets installed with `napari[all]`?

**Comment by melissawm** [comment]
Would love to explore that plugin for the docs, thanks for mentioning it @kephale !

**Comment by Czaki** [comment]
Also sphinx allow to build docs to qt docs format
